lib/sendgrid/recipient.rb

```diff
@@ -0,0 +1,26 @@
+require 'smtpapi'
+
+module SendGrid
+  class Recipient
+
+    attr_reader :address, :substitutions
+
+    def initialize(address)
+      @address = address
+      @substitutions = {}
+    end
+
+    def add_substitution(key, value)
+      substitutions[key.to_sym] = value
+    end
+
+    def add_to_smtpapi(smtpapi)
+      return if @address.nil? || @substitutions.empty?
```

Comment:
I feel more comfortable with objects that are always accessible only in a 'consistent' state, for some definition of consistent.  In this case, it seems like you can initialize this object and ask it to perform its main function and it will just fail silently.

I would probably set both address and substitutions on initialization and verify they are `present` there (though, no activesupport, so maybe `!nil`).